libraries/AC_PrecLand/AC_PrecLand.cpp

```diff
@@ -229,35 +251,51 @@ void AC_PrecLand::handle_msg(mavlink_message_t* msg)
 // Private methods
 //
 
+// initialise inertial buffer
+bool AC_PrecLand::init_inertial_buffer()
+{
+    // constrain lag parameter to be within bounds
+    _lag = constrain_float(_lag, 0.02f, 0.25f);
+
+    // calculate inertial buffer size from delay length (loop runs at main loop rate)
+    uint16_t inertial_buffer_size = MAX((uint16_t)roundf(_lag * AP::scheduler().get_loop_rate_hz()), 1);
```

Comment:
Did you mean MIN()?

Reply:
Nope, sorry.  My bad.

Reply:
.. although related to this, we should perhaps have a MIN around the get_loop-rate_hz() part.. something like this:

uint16_t inertial_buffer_size = MAX((uint16_t)roundf(_lag * MIN(AP::scheduler().get_loop_rate_hz(),400)), 1);

Reply:
If you min like that it doesn't work well on quadplanes at 300Hz in the future, which is part of the motivation for not hard coding it. And given that the buffer draining is based on call rate that's probably even worse then just allowing whatever the loop rate is to be used.

Reply:
@WickedShell, actually I think it would work for quadplanes (or any other vehicle).  We would put update_precland() into the scheduler with a desired update rate of 400hz.  If the main loop rate was less than 400hz, we would use the main loop rate, if it was over that, we would use 400hz.

As it is in this PR, if the loop rate is increased above 400hz it won't work properly.

Reply:
As loop rate is to soon increase, might be a good idea to try and work this out now?  What's the purpose of `MIN(AP::scheduler().get_loop_rate_hz(),400)` - why not AP::scheduler().get_loop_rate_hz()?
If the loop rate increases to something crazy like 8/16/32khz, will there be enough memory for the resulting buffer?

Reply:
@rmackay9 your right, it shoots faster copter loops in the foot. The best compromise is to actually pass the expected call rate in the scheduler table into the init() call, then do `MIN(expected_max_rate, scheduler_actual_max_loop_rate)`